mount/mount.go

```diff
@@ -13,6 +21,22 @@ type Mount struct {
 	Options []string
 }
 
+// Mounter handles mount and unmount
```

Comment:
This interface should stay where it is used. There is no justification for moving it as part of this PR.

Reply:
The createInitLayer has a similar problem, but the createInitLayer do mount by outside Mounter.
So for solving this, I moved Mounter into mount package.  It make the  function VisitMountedDir more general,

Reply:
Or else we only can solve it with defer for createInitLayer function.:)

Reply:
Another problem (very related to this) is that 6/7 callers pass `nil` for the `m Mounter` argument, which is not great. It doesn't read well.

I took a look at `createInitLayer()` and nothing uses it, or `InitRootFS()` or `Mounter` (I looked in moby and cri-containerd as well).

So for now how about keeping `Mounter` where it is (in `rootfs`), remove `type mounter struct`, and the `m Mounter` arg, and leaving `createInitLayer()` unmodified, with a comment about using `TempMount` when the Mounter interface is actually implemented ?

Reply:
@dnephin +1, having an interface there is premature. We are only keeping that function because of its potential use to a client, we haven't fully figured out if that init layers should just be completely deprecated or we need to support it in some way.

Reply:
ok, done